src/components/heading/heading.twig

```diff
@@ -20,15 +20,21 @@
   Build the class name based on the level.
 #}
 
-{% set _level_class = 'c-heading' %}
+{% set _heading_class = 'c-heading' %}
+{% set _heading_class_prefix = 'c-heading' %}
 
 {% if level is not empty %}
   {% set _level_modifier = ('level-' ~ level)|replace({ '--': '-n' }) %}
-  {% set _level_class = _level_class ~ ' ' ~ _level_class ~ '--' ~ _level_modifier %}
+  {% set _heading_class = _heading_class ~ ' ' ~ _heading_class_prefix ~ '--' ~ _level_modifier %}
+{% endif %}
+
+{% if weight is not empty %}
+  {% set _weight_modifier = 'c-heading--light' %}
+  {% set _heading_class = _heading_class ~ ' ' ~ _weight_modifier %}
```

Comment:
If you're going to check for the existence of `weight`, you should probably keep `_weight_modifier` as `_heading_class_prefix ~ '--' ~ weight`.